sympy/polys/polytools.py

```diff
@@ -2669,6 +2669,124 @@ def subresultants(f, g):
 
         return list(map(per, result))
 
+
+    def subresultant_polys(f, g):
+        """
+        Computes the subresultant polynomials of two polynomials ``f`` and ``g``.
+        The returned list has length min(deg(f), deg(g))+1, and is in order of
+        increasing degree. The first polynomial in the list is the resultant.
+
+        This uses the subresultant PRS. The remainder r_i is the deg(r_{i-1})-th
+        subresultant polynomial. Additionally, if deg(r_i) < deg(r_{i-1}) - 1,
+        then the deg(r_i) subresultant polynomial is r_i * LC(r_i) ^ c_i, where
+        c_i = deg(r_{i-1})-deg(r_i)-1.
+
+        Examples
+        ========
+
+        >>> from sympy import Poly
+        >>> from sympy.abc import x
+
+        >>> Poly(x**2 + 1, x).subresultant_polys(Poly(x**2 - 1, x))
+        [Poly(4, x, domain='ZZ'),
+         Poly(-2, x, domain='ZZ'),
+         Poly(x**2 - 1, x, domain='ZZ')]
+
+        Observe that the first element is the resultant.
+
+        >>> Poly(x**2 + 1, x).resultant(Poly(x**2 - 1, x))
+        4
+
+        """
+
+        # ensure deg(f) >= deg(g)
+        n = f.degree()
+        m = g.degree()
+
+        if n < m:
+            f, g = g, f
+            n, m = m, n
+
+        prs = f.subresultants(g)
+        if len(prs) <= 1:
+            return []
+
+        subres_polys = [0] * (m + 1)
+
+        for i in reversed(range(2, len(prs))):
+            # remainder r_i is the deg(r_{i-1})-th subres poly
+            subres_polys[ prs[i-1].degree() -1 ] = prs[i]
+
+            # if there is a "degree jump"
+            # if deg(r_i) < deg(r_{i-1}) - 1, then the deg(r_i) subres poly
+            # is r_i * LC(r_i) ^ c_i, where c_i = deg(r_{i-1})-deg(r_i)-1
+            if prs[i].degree() < prs[i-1].degree() - 1:
+                degree_jump = prs[i-1].degree() - prs[i].degree() - 1
+                subres_polys[ prs[i].degree() ]  =\
+                    prs[i] * prs[i].LC() ** degree_jump
+
+        # handle last one
+        subres_polys[-1] = prs[1] * g.LC() ** (n - m - 1)
+
+        # want it to always return Poly objects for consistency
+        # this basically just makes sure 0 is returned as Poly(0)
+        for i in range(len(subres_polys)):
+            if not isinstance(subres_polys[i], Poly):
+                subres_polys[i] = Poly(i, f.gen)
```

Comment:
I think this code could be simplified by using a single loop that appends items one at a time.